### ISSUE TITLE
Include more unpaved highways in smoothness quest

### ISSUE THREAD

**Issue by rhhsm** [streetcomplete/StreetComplete#5499]
**Use case**
At the moment, most unpaved highways such as tracks are not included in the smoothness quest because their surface is often `ground`, `dirt`, `earth`, etc. I think that for car users, these ways are especially interesting to be tagged with smoothness because they are often worse than `bad` and it's at this level of smoothness where it really matters which kind of car you have for a road to be passable or not.

**Proposed Solution**
I think ways with `ground`, `dirt`, `earth` and even `unpaved` surfaces can be included in the smoothness quest by using the images now used for `surface=gravel`. I think the differences between these surfaces are not very clear from the wiki anyway, so the same images can be used for all of them. The images now used for `very_bad`, `horrible` and `very_horrible` are mine and are of tracks that I tagged with `surface=ground/dirt`.

**Comment by FloEdelmann** [comment]
For reference: https://github.com/streetcomplete/StreetComplete/issues/5325#issuecomment-1772637797

**Comment by rhhsm** [comment]
> For reference: [#5325 (comment)](https://github.com/streetcomplete/StreetComplete/issues/5325#issuecomment-1772637797)

You mean to refer to "**The only thing that is missing are fitting smoothness-pictures for dirt/ground roads.**"? I think we are already have them, but now we're using them for `surface=gravel` only. The images for `very_bad` and `horrible` show tracks that are actually tagged with `surface=ground` so surely we can use those images for `surface=ground`-tracks too. The `very_horrible` picture sows a track with `surface=dirt`.

I do agree that some of the pictures shown in the smoothness gallery are problematic, so I intend to start a discussion in the forum as soon as I have some time for it.

**Comment by matkoniecz** [comment]
From looking at code: we have right now

```
private val Smoothness.compactedImageResId get() = when (this) {
    GOOD -> R.drawable.surface_compacted_good
    INTERMEDIATE -> R.drawable.surface_compacted_intermediate
    BAD -> R.drawable.surface_compacted_bad
    VERY_BAD -> R.drawable.surface_compacted_very_bad
    HORRIBLE -> R.drawable.surface_unpaved_horrible
    VERY_HORRIBLE -> R.drawable.surface_unpaved_very_horrible
    IMPASSABLE -> R.drawable.surface_unpaved_impassable
    else -> null
}

private val Smoothness.gravelImageResId get() = when (this) {
    INTERMEDIATE -> R.drawable.surface_gravel_intermediate
    BAD -> R.drawable.surface_gravel_bad
    VERY_BAD -> R.drawable.surface_gravel_very_bad
    HORRIBLE -> R.drawable.surface_unpaved_horrible
    VERY_HORRIBLE -> R.drawable.surface_unpaved_very_horrible
    IMPASSABLE -> R.drawable.surface_unpaved_impassable
    else -> null
}

private val Smoothness.compactedOrGravelDescriptionResId get() = when (this) {
    GOOD -> R.string.quest_smoothness_description_good_compacted_gravel
    INTERMEDIATE -> R.string.quest_smoothness_description_intermediate_compacted_gravel
    BAD -> R.string.quest_smoothness_description_bad_compacted_gravel
    VERY_BAD -> R.string.quest_smoothness_description_very_bad_compacted_gravel
    else -> null
}
```

So we would probably need values for `good` `intermediate` and maybe `bad` and `very_bad` if current ones do not fit.

> You mean to refer to "The only thing that is missing are fitting smoothness-pictures for dirt/ground roads."? 

Yes

> I think we are already have them, but now we're using them for surface=gravel only.

Are you sure that all current gravel images fit also for dirt? I can check if you are sure.

Though note

> And in either case, these things should be contributed to the wiki first
( https://wiki.openstreetmap.org/wiki/Key:smoothness/Gallery ) and discussed, if necessary, in the forum, not in this issue tracker.

**Comment by rhhsm** [comment]
> So we would probably need values for `good` `intermediate` and maybe `bad` and `very_bad` if current ones do not fit.
> 
I don't think we need to show `good` (we don't show it for `gravel`  surfaces).

> Are you sure that all current gravel images fit also for dirt? I can check if you are sure.
> 
I think so, yes, though that might be controversial. The wiki only shows pictures for asphalt and unpaved roads and that seems to work fine.

**Comment by matkoniecz** [comment]
https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/res/drawable-xhdpi/surface_gravel_intermediate.jpg - for example this is clearly not `surface=dirt`

Closing per

> And in either case, these things should be contributed to the wiki first
( https://wiki.openstreetmap.org/wiki/Key:smoothness/Gallery ) and discussed, if necessary, in the forum, not in this issue tracker.

**Comment by matkoniecz** [comment]
And finding good images for these cases would be highly welcomed!

**Comment by rhhsm** [comment]
> https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/res/drawable-xhdpi/surface_gravel_intermediate.jpg - for example this is clearly not `surface=dirt`

But can it be used to illustrate `ground`? Or alternatively, maybe this one could be used https://wiki.openstreetmap.org/wiki/File:Smoothness_intermediate_unpaved.jpg ? Or this one https://wiki.openstreetmap.org/wiki/File:Rainy_dirt_road_on_the_Tibetan_Plateau.jpg ?

Why can https://wiki.openstreetmap.org/wiki/File:Very_horrible_ruts.jpg be used to illustrate `very_horrible`
 `gravel` though it is clearly not `surface=gravel`?

**Comment by matkoniecz** [comment]
> Why can https://wiki.openstreetmap.org/wiki/File:Very_horrible_ruts.jpg be used to illustrate very_horrible
gravel though it is clearly not surface=gravel?

Where it is used for this purpose (sorry if I missed it somehow, posting from smartphone in bus) - but I do not really see it at relevant gallery at Wiki being used for gravel

**Comment by rhhsm** [comment]
> > Why can https://wiki.openstreetmap.org/wiki/File:Very_horrible_ruts.jpg be used to illustrate very_horrible
> > gravel though it is clearly not surface=gravel?
> 
> Where it is used for this purpose (sorry if I missed it somehow, posting from smartphone in bus) - but I do not really see it at relevant gallery at Wiki being used for gravel

In  StreetComplete, when solving a smoothness quest for a way with `surface=gravel`. It's a tu quoque argument, however.

![Screenshot](https://github.com/streetcomplete/StreetComplete/assets/69100427/8b605766-e232-4dcf-9174-38463162f2cb)

What I'd like to say is that we can and should be a lot less demanding for pictures that show a smoothness level for a certain kind of unpaved surface because these surfaces are not very well defined in the key:surface wiki anyway. The key point is that they demonstrate a level of smoothness; what surface they show is less important

**Comment by mnalis** [comment]
> The key point is that they demonstrate a level of smoothness; what surface they show is less important

For OSM wiki I might perhaps agree, but as SC has specially tailored set of answers for each type of surface, I disagree.

Especially as SC is tailored towards **newbie users, who have no idea of OSM tags** and their definition.  Which is why SC shows different sets of pictures for different surfaces.

If it e.g. asks them **what is the quality of the asphalt surface, and shows picture of gravel** (or gravel vs. dirt, or whatever), they will be **highly confused**. IMHO, such confusion should be avoided in SC, even if it means asking less questions (on the other hand, for [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) full range of surface+smoothness perhaps might work, even without pictures where there don't exist any; but not in regular SC)

So, I think that way to proceed with [smoothness quest](https://github.com/streetcomplete/StreetComplete/blob/2aa9bb6ccc3fc974636581166fc935e76404de4b/app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddRoadSmoothness.kt#L49-L51) being asked on `surface ~ dirt|earth|ground` in StreetComplete too, is to find valid pictures for every `surface`-`smoothness` pair which depicts that situation (including fixing current pictures which are problematic, as well as finding new ones for new surfaces).

> You mean to refer to "The only thing that is missing are fitting smoothness-pictures for dirt/ground roads."? I think we are already have them, but now we're using them for surface=gravel only. The images for very_bad and horrible show tracks that are actually tagged with surface=ground so surely we can use those images for surface=ground-tracks too. The very_horrible picture sows a track with surface=dirt.

Well, yeah, that picture is not good, and it should be replaced IMHO. If there are real examples of `horrible` `gravel`, someone should find picture of that. It is probably just that it is pretty rare so no one has been able to take or find freely licensed picture it yet, so someone added this less-then-ideal picture... Did anyone ever seen such surface? Could they take a picture of that? That would improve the current situation, IMHO.  If not, perhaps we could ask some graphic AI to generate one for us (although, that might bring copyright status uncertainty), or even remove that answer if it never could happen? 

> I do agree that some of the pictures shown in the smoothness gallery are problematic, so I intend to start a discussion in the forum as soon as I have some time for it.

If/when you do, please also link here so interested parties can follow too!

**Comment by rhhsm** [comment]
Thanks @mnalis for your post. Indeed we should confuse the newbies as little as possible, but I'm afraid we're doing it now with some of the pictures already accepted for use in SC so it feels a bit unfair to me that what I propose here is seen as too confusing... (but feelings should not play a role here).
Indeed showing a gravel surface when asking about an asphalt-paved road will be confusing, but showing a ground surface when asking about a gravel-paved road is much less so because ground and gravel are very similar surfaces (what one mapper will map as `gravel`, another will easily map as `ground`). Even a dirt-road can be shown, I think, because on the picture it will hardly be noticeable that the small stones that `gravel` should have are absent.

> Well, yeah, that picture is not good, and it should be replaced IMHO. If there are real examples of `horrible` `gravel`, someone should find picture of that. It is probably just that it is pretty rare so no one has been able to take or find freely licensed picture it yet, so someone added this less-then-ideal picture... Did anyone ever seen such surface? Could they take a picture of that? That would improve the current situation, IMHO. If not, perhaps we could ask some graphic AI to generate one for us (although, that might bring copyright status uncertainty), or even remove that answer if it never could happen?

Maybe it would be better to show an icon for "no picture available" than a confusing picture? Or even a text saying "We don't have a picture for this, do you have one? Please upload it to https://wiki.openstreetmap.org/wiki/Key:smoothness/Gallery " 
> 
> > I do agree that some of the pictures shown in the smoothness gallery are problematic, so I intend to start a discussion in the forum as soon as I have some time for it.
> 
> If/when you do, please also link here so interested parties can follow too!

I started a thread for general discussion of `smoothness` images here https://community.openstreetmap.org/t/how-to-map-key-smoothness/109860
I also added another table to https://wiki.openstreetmap.org/wiki/Key:smoothness/Gallery to show pictures for unpaved surfaces, moved the `gravel/fine_gravel` row there and added ones for `ground` and `dirt/earth`. I will be looking for more pictures to fill the gaps, and will post on the comminity forum to ask for help and comments

Could someone re-open this issue so it becomes more visible?

**Comment by matkoniecz** [comment]
> Thanks @mnalis for your post. Indeed we should confuse the newbies as little as possible, but I'm afraid we're doing it now with some of the pictures already accepted for use in SC so it feels a bit unfair to me that what I propose here is seen as too confusing... (but feelings should not play a role here).

Note that there is possibility that current `gravel` support should be disabled until we will find better pictures and maybe you will convince us (though that should be a new separate issue). But `dirt` will be definitely not added with clearly `gravel` pictures.

> Could someone re-open this issue so it becomes more visible?

See https://github.com/streetcomplete/StreetComplete/issues/5325#issuecomment-1775781955 and https://github.com/streetcomplete/StreetComplete/issues/5325#issuecomment-1772637797

Currently this requires no coding as images are needed first

> And in either case, these things should be contributed to the wiki first ( https://wiki.openstreetmap.org/wiki/Key:smoothness/Gallery ) and discussed, if necessary, in the forum, not in this issue tracker.

**Comment by RubenKelevra** [comment]
What do you think about generating the missing images? 

I created an example set for ground / dirt / earth:

### Good
<img src="https://github.com/streetcomplete/StreetComplete/assets/614929/f87e8bb8-d581-4892-ad92-d03196137391" width="30%">

### Intermediate
<img src="https://github.com/streetcomplete/StreetComplete/assets/614929/fd89b523-82c5-44b2-92e0-9af0ad1b5ddb" width="30%">

### Bad
<img src="https://github.com/streetcomplete/StreetComplete/assets/614929/0696577f-2d59-436c-8978-48caa003c74c" width="30%">

### Very_bad
<img src="https://github.com/streetcomplete/StreetComplete/assets/614929/b3c0e5c6-dfd1-4752-8806-39db5951ad20" width="30%">

### Horrible
<img src="https://github.com/streetcomplete/StreetComplete/assets/614929/3d971b64-8431-497b-bc60-7531290cc93f" width="30%">

### Very_horrible
<img src="https://github.com/streetcomplete/StreetComplete/assets/614929/f4917eed-53ab-4314-bd2e-f156414c98ae" width="30%">

### Impassable
<img src="https://github.com/streetcomplete/StreetComplete/assets/614929/a2a29668-01b8-4560-83be-bf199291334f" width="30%">

**Comment by RubenKelevra** [comment]
Hey @westnordost what's your opinion on that? The images above are just an example and are certainly not the best this approach can deliver.

**Comment by westnordost** [comment]
Hm, I am not sure what's my stand on AI generated images. I think good images could be produced with that, but these look... cool, yes, but hyperrealistic, i.e. kind of not like normal photos, like from some kind of artsy photo shoot. This clashes somewhat with the other pictures. Regarding the classification, everything down from very bad looks very similar, unfortunately. (The same with bad, intermediate, good, IMO)

**Comment by RubenKelevra** [comment]
Thanks for the feedback. Yeah I was going with my standard approach to describe how images should look like, so it ended up more like an art image than a descriptive one.

That's why I put in that disclaimer regarding "not optimal".

I agree that the images look similar, but they follow how I understand those categories, meaning the difference between `very_bad` and `horrible` is "SUV" vs dedicated offroad vehicle.

And between `horrible` and `very_horrible` is offroad vehicle to tractor.

`impassable` and `good` actually don't not exist in the wiki as photo row, and I it took considerable effort to generate those images (and I'm not even happy with the results).

So I kind of agree with you in this point.

---

So question is @westnordost / @matkoniecz / @rhhsm, do we go forward with this idea?

I takes around 1-2 hours to create such a set, maybe a bit less if I skip the best and worst options, as they are very difficult to generate.

I would then put them in the wiki with CC-0, as AI "art" cannot be legally copyrighted in the EU, and we could use them to get tracks smoothness to work with more surfaces (finally!) 😬

If you don't wanna go ahead, this time effort doesn't make much sense. So please advice.

**Comment by rhhsm** [comment]
Sorry I haven't commented until now. I've been working on this and on aligning the various interpretations of `smoothness` in general ([here](https://community.openstreetmap.org/t/categorisation-of-key-smoothness-values/110063/28) and [here](https://community.openstreetmap.org/t/how-to-map-key-smoothness/109860/50)). I started a sandbox page [here](https://wiki.openstreetmap.org/wiki/User:Rhhsmits/SmoothnessGallery) where I worked on a new version of the smoothness gallery, but haven't found the time to complete it. I also made pictures while walking around my wife's village (very rich collection of surfaces in various states of smoothness!) and started to sort them according to what I think is their smoothness category. I just uploaded them [here](https://drive.google.com/drive/folders/1Y4Q7gQN_gp315K3T6L2J93wg95ZQpjfq?usp=sharing).

@RubenKelevra maybe we can cooperate on this?

I think suitability of pictures to illustrate `smoothness` is of concern to OSM in general so should be discussed on the community forum, and then adopted by SC

**Comment by mnalis** [comment]
Thanks @RubenKelevra for the effort!

While I'm not sure that AI copyright status (in EU or elsewhere) is anywhere near concluded, I think that we should **if at all possible expend all effort to try to find real-life examples**. Only if that turns out to be unworkable, should we  try alternatives:

- delaying quest expansion indefinitely until we can find such pictures (well, that's a default anyway)
- accept quest expansion, but omit values for "impossible to find pictures" (as probably quite rare, let users leave notes)
- accept quest expansion, show all values (with "?" pictures for ones that we are missing pics), and when those values are selected instead (or in addition) to tagging the value, ask the user to please leave a picture in a note so we can use it for future SC releases (so perhaps something useful pops up).
- search where those "hard to find" values were used recently, and ask changesets authors or other nearby users if they could take pictures of it for OSM/SC? Or going there ourselves if it happens not be too far.
- (...shortened...) I don't think we should use AI even for that one "almost impossible" picture only, instead we should be preferring any other alternative above -- AI is not only problematic from legal landscape, but much more so **social one** -- i.e. many people might ragequit or worse if they got the idea that SC is _"implicitly supporting AIs and thus getting artists laid off and replaced by AIs"_; and I do not see those tensions getting any weaker as time passes by, quite the opposite. Just does not seem worth the potential risk.

**Comment by RubenKelevra** [comment]
@mnalis I don't like the approach you've outlined.

I think it's fine to use AI images as default set and replace the images we got real photos from, as soon as they are available.

I think that's the better approach, as these images are tiny and users won't really notice anyway.

AI image copyrights are settled - there is none. Please provide evidence if you wanna argue against this.

Here is what I read before making my claim, that there's no copyright for AI art:

"Juristen sind sich einig, dass KI-generierte Werke im Normalfall keinen Urheber im rechtlichen Sinne haben, sagt Rechtsanwalt Joerg Heidrich, der auch KI-Unternehmen berät: "Für den Nutzer heißt das, er kann ein KI-generiertes Bild frei verwenden und es etwa auf seine Webseite stellen - auf der anderen Seite kann das jeder andere mit dem gleichen Bild aber auch tun."

Source: https://www.google.com/amp/s/amp.zdf.de/nachrichten/wirtschaft/urheberrecht-kuenstliche-intelligenz-ki-internet-100.html

"The US Copyright Office recently ruled again that artworks created by artificial intelligence are not protected by copyright, upholding its practice of not registering AI-generated art as a copyright work."

https://haerting.de/en/insights/usa-no-copyright-for-ai-generated-images/

**Comment by RubenKelevra** [comment]
> @RubenKelevra maybe we can cooperate on this?

Sure! What do you need me to do? 

> I think suitability of pictures to illustrate `smoothness` is of concern to OSM in general so should be discussed on the community forum, and then adopted by SC

Yeah I don't like doing a long discussion about that with the community. That will set us back for month without any value added. If you wanna discuss this, fine. But then I'm out.

I would just create the images, add them to the grid there as default if nothing else exist and use them in SC until some man made image is added.

Issue solved.

**Comment by mnalis** [comment]
**TL;DR:** I don't think AI legal situation is nowhere near "solved", nor will likely be for at least few years. Even more so, I think negative publicity of using AI is important consideration (despite perhaps seeming like short-term win). Thus, I suggest considering  [previously suggested](https://github.com/streetcomplete/StreetComplete/issues/5499#issuecomment-2201954455) alternatives.

----

> I don't like the approach you've outlined.

Sure, that's fine! Neither do I particularly like AI approach :smiley_cat: , but I've tried to explain _why_ I don't like it, and _what_ alternatives / compromises I *do* see. Would you care to say what you like/dislike about each of the 5 proposals I've suggested earlier? Or propose your own alternatives?
 
> as these images are tiny and users won't really notice anyway.

I think it's bad idea to do something that one knows/suspects is legally/socially problematic, and then hope people won't notice because it's tiny / hard to spot. 

> "The US Copyright Office recently ruled again that artworks created by artificial intelligence are not protected by copyright, upholding its practice of not registering AI-generated art as a copyright work."

- You seem to be talking about _"AI output cannot be copyrighted"_, which is **irrelevant** here. If it cannot be copyrighted, it is public domain anyway. But even if it could be copyrighted, it would likely be up to the party running AI to assign copyright (i.e. You), and you could easily put it under CC0 -- thus legal the result for SC would be the same (allowed use of PD/CC0 image). Thus, it is not an issue for SC.

- However, much more contested  question is _"Are you even allowed to train AI on dataset whose copyright does not explicitly allow that?"_ - and for that I would say the **jury is still out**. See for example [Getty Images vs. Stability AI](https://cdn.vox-cdn.com/uploads/chorus_asset/file/24412807/getty_images_vs_stability_AI_delaware.pdf) -- has it been resolved in court already? If not, then the resolution is still unknown, yes? And if it is illegal, then output is also illegal (regardless whether the output could otherwise be copyrighted or not).

Also, world is much bigger than USCO, and jurisdiction's interpretations vary. I think we will have to have at least few dozen (if not more) court cases resolved (through all appeals) in various jurisdictions before anyone could give some reasonable judgement. 

If your AI had been trained **only** on your images (which you released via e.g. CC0) and then you run the new image generation using **only** that training dataset, then the legal issue would be clear if you also release result as CC0. But if you're (like the most of the rest of the world) using pre-processed model, then the legality of the output might depend on what images that model was trained on.

---

But as I've tried to emphasize above, **social animosity towards AI** are even bigger issue, and that one is threatening to become unstoppable forest fire; you can barely find a week when there is not at least one of few new outbursts, e.g.:

-  2024-06-07 [Adobe stealing user data to train their AI](https://www.theverge.com/2024/6/7/24173838/adobe-tos-update-firefly-generative-ai-trust)
- 2024-07-03 [Cloudflare to block AI scrapers accessing customer data](https://www.theverge.com/2024/7/3/24191698/cloudflare-is-offering-to-block-crawlers-scraping-information-for-ai-bots)
- 2024-06-28 [Microsoft’s AI boss thinks it’s perfectly okay to steal content if it’s on the open web](https://www.theverge.com/2024/6/28/24188391/microsoft-ai-suleyman-social-contract-freeware)
-  2024-06-26 [An AI-designed horse purse is tearing apart this small but passionate community](https://www.theverge.com/2024/6/26/24186635/baggu-collina-strada-collaboration-ai-purses-art)
- 2024-04-02 [celebrities sign Open Letter Warning About AI Threat to Artists](https://www.hollywoodreporter.com/business/business-news/billie-eilish-nicki-minaj-open-letter-ai-1235864532/)
- 2024-06-21 [OpenAI exec: “Some creative jobs maybe will go away, but maybe they shouldn’t have been there in the first place.”](https://www.theverge.com/2024/6/21/24183265/openai-exec-some-creative-jobs-maybe-will-go-away-but-maybe-they-shouldnt-have-been-there-in-the-fir)
- 2024-06-13 [Clearview AI Used Your Face. Now You May Get a Stake in the Company. ](https://www.nytimes.com/2024/06/13/business/clearview-ai-facial-recognition-settlement.html)
- 2024-06-13 [Picsart teams up with Getty to take on Adobe’s ‘commercially-safe’ AI](https://www.theverge.com/2024/6/14/24178333/picsart-getty-commercially-safe-ai-image-tool-adobe)

**Comment by rhhsm** [comment]
It seems to me that using AI generated instead of volunteer-generated images to illustrate `smoothness` is not an ethical issue (just completed an "Ethics Appraisal" training at the EC), as long as it's not claimed that the images represent reality. There might be a legal issue for @RubenKelevra if the AI was trained using copyrighted material without permission...

I think we already have better volunteer-generated images for most cases ;)

**Comment by mnalis** [comment]
> I think we already have better volunteer-generated images for most cases ;)

"most" meaning "not all", right? Which ones are missing (e.g. for mentioned `ground` / `earth` / `dirt`)?

I'm all for it that we humans go acquire few missing ones; if everything but pictures is ready for improved quest.

**Comment by matkoniecz** [comment]
Some minor coding is needed but images are main barrier

**Comment by matkoniecz** [comment]
I have taken some but I need to look how well they work for illustrating smoothness

**Comment by matkoniecz** [comment]
https://github.com/streetcomplete/StreetComplete/blob/7c04687816c9e455ccedce7858f61cb0d59b6d5d/app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddRoadSmoothness.kt#L50

So currently `"asphalt", "concrete", "concrete:plates", "sett", "paving_stones", "compacted", "gravel", "fine_gravel"` are supported

**Comment by RubenKelevra** [comment]
@matkoniecz thanks for the bump!

@rhhsm have you an update? :)

**Comment by rhhsm** [comment]
No, and I won't have much time to work on it until December 

 
 
  Op wo, sep 18, 2024 om 8:09 schreef @***@***.***>:   


@matkoniecz thanks for the bump!

@rhhsm have you an update? :)
